src/internal/connector/exchange/attachment.go

```diff
@@ -46,54 +47,52 @@ func uploadAttachment(
 	uploader attachmentUploadable,
 	attachment models.Attachmentable,
 ) error {
-	logger.Ctx(ctx).Debugf("uploading attachment with size %d", *attachment.GetSize())
+	attachmentType := attachmentType(attachment)
+
+	ctx = clues.AddAll(
+		ctx,
+		"attachment_size", ptr.Val(attachment.GetSize()),
+		"attachment_id", ptr.Val(attachment.GetId()),
+		"attachment_name", ptr.Val(attachment.GetName()), // TODO: pii
+		"attachment_type", attachmentType,
+		"internal_item_type", getItemAttachmentItemType(attachment),
+		"uploader_item_id", uploader.getItemID())
+
+	logger.Ctx(ctx).Debugw("uploading attachment")
 
-	var (
-		attachmentType = attachmentType(attachment)
-		err            error
-	)
 	// Reference attachments that are inline() do not need to be recreated. The contents are part of the body.
-	if attachmentType == models.REFERENCE_ATTACHMENTTYPE &&
-		attachment.GetIsInline() != nil && *attachment.GetIsInline() {
-		logger.Ctx(ctx).Debugf("skip uploading inline reference attachment: ", *attachment.GetName())
+	if attachmentType == models.REFERENCE_ATTACHMENTTYPE && ptr.Val(attachment.GetIsInline()) {
+		logger.Ctx(ctx).Debug("skip uploading inline reference attachment: ", ptr.Val(attachment.GetName()))
 		return nil
 	}
 
 	// item Attachments to be skipped until the completion of Issue #2353
 	if attachmentType == models.ITEM_ATTACHMENTTYPE {
-		prev := attachment
-
-		attachment, err = support.ToItemAttachment(attachment)
+		a, err := support.ToItemAttachment(attachment)
 		if err != nil {
-			name := ptr.Val(prev.GetName())
-			msg := "item attachment restore not supported for this type. skipping upload."
-
-			// TODO: (rkeepers) Update to support PII protection
-			logger.Ctx(ctx).Infow(msg,
-				"err", err,
-				"attachment_name", name,
-				"attachment_type", attachmentType,
-				"internal_item_type", getItemAttachmentItemType(prev),
-				"attachment_id", ptr.Val(prev.GetId()),
-			)
+			logger.Ctx(ctx).
+				With("err", err).
```

Comment:
nit: I don't think we need to worry about number of characters output to the log so perhaps "error" would be better? ;) We should also probably make a standard for this field cause it will be rather common in our log output so it should be made easy to search for

```suggestion
				With("error", err).
```

Reply:
I've seen both cases be standard.  Let's put a pin on this and I can ensure everything uses the same format as a cleanup at the end.